java-squid/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -529,18 +529,20 @@ private boolean isArgumentsAcceptable(List<JavaType> argTypes, List<JavaType> fo
   }
 
   private boolean isAcceptableType(JavaType arg, JavaType formal, boolean autoboxing, boolean isParameterized) {
-    // FIXME Inference of types for method using parameter types is not handled so fall back to behavior based on erasure
-    if (!isParameterized && usesWildcard(formal)) {
+    // FIXME SOANRJAVA-1514 Inference of types for method using parameter types is not handled
+    // FIXME SONARJAVA-1535 type substitution should work with wildcards
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)